packages/page-contracts/src/Contracts/Deploy.tsx

```diff
@@ -50,6 +51,7 @@ function Deploy ({ codeHash, constructorIndex = 0, onClose, setConstructorIndex
 
   const [name, isNameValid, setName] = useNonEmptyString(code && code.json.name);
   const { contractAbi, errorText, isAbiError, isAbiSupplied, isAbiValid, onChangeAbi, onRemoveAbi } = useAbi([code && code.json.abi, code && code.contractAbi], codeHash, true);
+  const isConstructorPayable = contractAbi?.constructors[constructorIndex].isPayable !== false;
```

Comment:
the intention is to make the constructors explicitly payable / not payable in ink! 
just trying to be forward compatible. but yes, it's an unnecessary check at this point.

Reply:
I understand that, but it is actually not forward-compatible at all :) So, if we want to really do this - pass the flag here https://github.com/polkadot-js/api/blob/master/packages/api-contract/src/Abi/index.ts#L93 (like we do for messages)

For now it would need to always be true. When the time comes that the constructors pass it, on the API level it would need to change into an `Option<bool>` for that fields where `isPayable: o.isNone || o.unwrap().isTrue` (to maintain backwards compat)

So not against the checks, however the tristate logic should not sit quite here. It is ok for now, but needs cleanup on the ABI level and then here.